### PR TITLE
added minimal test for DD, DT neutron energy

### PR DESCRIPTION
**Pull request by shimwell**
This PR adds some minimal tests for the DTmean and DDmean neutron energy.

This is in preparation and to protect functionality for a follow up PR where I try to change the units to eV and I will want to show that things are still working.

the ci.yml file has also been changed so that these 4 tests run in the CI

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    DTmean, _ = nst.DTprimspecmoments(Tion=5.0)  # units KeV
+
+    assert DTmean == pytest.approx(14.1, abs=0.1)  # units MeV
+
+
+def test_DDprimspecmoments_mean():
+    # checks the mean value of the neutron emitted by DD fusion
+
+    DTmean, _ = nst.DDprimspecmoments(Tion=5.0)  # units KeV
```
Comment (shimwell) [inline tests/test_core.py:16]:
```suggestion
    DDmean, _ = nst.DDprimspecmoments(Tion=5.0)  # units KeV
```

File: tests/test_core.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    assert DTmean == pytest.approx(2.5, abs=0.1)  # units MeV
```
Comment (shimwell) [inline tests/test_core.py:18]:
```suggestion
    assert DDmean == pytest.approx(2.5, abs=0.1)  # units MeV
```

File: tests/test_core.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    assert DTmean == pytest.approx(14.1, abs=0.1)  # units MeV
+
+
+def test_DDprimspecmoments_mean():
+    # checks the mean value of the neutron emitted by DD fusion
+
+    DDmean, _ = nst.DDprimspecmoments(Tion=5.0)  # units KeV
+
+    assert DDmean == pytest.approx(2.5, abs=0.1)  # units MeV
+
```
Comment (shimwell) [inline tests/test_core.py:19]:
```suggestion


```

File: tests/test_core.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+def test_DDprimspecmoments_mean_with_tion():
+    # checks the energy of the neutron increases with ion temperature
+
+    DDmean_cold, _ = nst.DDprimspecmoments(Tion=5.0)  # units KeV
+    DDmean_hot, _ = nst.DDprimspecmoments(Tion=10.0)  # units KeV
+
+    assert DDmean_cold < DDmean_hot  # units MeV
+
```
Comment (shimwell) [inline tests/test_core.py:27]:
```suggestion


```

**Review by aidancrilly — APPROVED** [review]
Just approving at this stage, not sure if you are adding more tests but these seem sensible to me

**Comment by shimwell** [comment]
I'm keen to do more tests at a later date. I'm sure I'll get more ideas while shifting the units to eV. Is it ok to merge this one

**Comment by aidancrilly** [comment]
Happy to merge

**Pull request merged**